wenet/transducer/transducer.py

```diff
@@ -216,27 +216,30 @@ def greedy_search(
         state_m, state_c = self.predictor.init_state(1, method="zero")
         t = 0
         hyps = []
-        prev_out_nblk = False
+        prev_out_nblk = True
         pred_out_step = None
-        per_frame_max_noblk = 5
+        per_frame_max_noblk = 100
         per_frame_noblk = 0
         while t < encoder_out_lens:
             encoder_out_step = encoder_out[:, t:t + 1, :]  # [1, 1, E]
-            if not prev_out_nblk:
-                pred_out_step, state_m, state_c = self.predictor.forward_step(
+            if prev_out_nblk:
+                pred_out_step, state_out_m, state_out_c = self.predictor.forward_step(
```

Comment:
We can encode the state into one tensor in the future so it be applied to any kind of predictors. Here `m` and `c` is for LSTM like predictor.

Reply:
yes， i will implement embedding predictor after aishell works， and then concat into one state